kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -135,12 +127,12 @@ sealed abstract class Stream[V, -S]:
             Stream[V, S](ArrowEffect.handleState(tag, n, emit)(
                 [C] =>
                     (input, state, cont) =>
-                        if state == 0 then
-                            (0, cont(Stop))
-                        else
-                            val c   = input.take(state)
-                            val nst = state - c.size
-                            Emit.andMap(c)(ack => (nst, cont(ack.maxValues(nst))))
+                        debug.Debug((state, input)).andThen:
+                            if state == 0 then (0, Kyo.pure[Unit, Emit[Chunk[V]] & S](()))
```

Comment:
@fwbrasil does this make sense? When the state reaches zero, I don't call `cont` at all but just return unit. This seems right to me and works, but I haven't found examples elsewhere of short-circuiting this way. In fact, I can't figure out how the old version of `Emit` accomplished early termination via `Stop` (since all the handlers, as far as I can tell, always pass `Stop` to `cont`, which would presumably keep running the effect, even if it no longer emitted anything.

Reply:
Yes, the previous implementation would let the computation finish and the emitter was responsible for stopping. Short-circuiting by discarding the continuation is ok. It think it only happens in `Abort` and `Choice` for now (and derived effects like `Parse`). There's a mechanism that attaches finalizers to the fiber so they're cleared up even if the continuation isn't evaluated.